providers/fips/fipsprov.c

```diff
@@ -904,6 +905,10 @@ int OSSL_provider_init_int(const OSSL_CORE_HANDLE *handle,
     if (!ossl_thread_register_fips(libctx))
         goto err;
 
+    /* Ensure our internal provider is loaded */
```

Comment:
I would prefer to see a more detailed comment here explaining why this is essential:

* The internal provider is needed when the FIPS module uses the EVP API to ultimately again call functions inside the FIPs provider.
* That sort of call is about to happen (briefly note where/how) and we need to have the internal provider ready now.

Reply:
I expanded out the comment.